homeassistant/components/garage_door/myq.py

```diff
@@ -0,0 +1,172 @@
+"""
+Support for Chamberlain garage doors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/garage_door.myq/
+"""
+
+import logging
+import requests
+
+from homeassistant.components.garage_door import GarageDoorDevice
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_NAME
+
+
+REQUIREMENTS = []
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the MyQ garage door."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    name = config.get(CONF_NAME) if \
+        CONF_NAME else "MyQ"  # is this the right way?
+
+    if username is None or password is None:
+        logging.getLogger(__name__).error(
+            "Missing username or password.")
+        return
+
+    myq = MyQAPI(username, password)
```

Comment:
How do you ensure that the username/password combination are valid and accepted? Usually catching a communication error would be appreciated because otherwise the user is getting a platform which is not working.